client/src/components/Input/AddressInput.js

```diff
@@ -0,0 +1,106 @@
+import axios from "axios";
+import React, { useState } from "react";
+import styled from "styled-components";
+import { Dropdown, Menu } from "antd";
+import { v4 as uuidv4 } from "uuid";
+
+import Input from "components/Input/BaseInput";
+import InputError from "components/Input/InputError";
+import { inputStyles } from "constants/formStyles";
+import { theme } from "constants/theme";
+
+const StyledMenu = styled(Menu)`
+  padding: ${theme.typography.size.xxsmall};
+  width: 100%;
+  .ant-dropdown-menu-item {
+    display: flex;
+    span {
+      font-size: ${theme.typography.size.medium};
+      margin-right: ${theme.typography.size.xxsmall};
+      :first-child {
+        color: ${theme.colors.darkerGray};
+        font-weight: 500;
+      }
+      :last-child {
+        margin: 0;
+      }
+    }
+    &.ant-dropdown-menu-item-active {
+      background-color: ${theme.colors.selago};
+    }
+  }
+`;
+
+const SubLabel = styled.small`
+  color: ${props => props.selected ? theme.colors.lightGray : theme.colors.green};
+  display: block;
+`;
+
+const AddressInput = ({ location, errors, onLocationChange }) => {
+  // sessiontoken for combining autocomplete & palce details into single usage
+  // see: https://developers.google.com/maps/billing/gmp-billing#ac-with-details-session
+  const [geoSessionToken, setGeoSessionToken] = useState(uuidv4());
+  const [predictedAddresses, setPredictedAddresses] = useState([]);
+  const [inputAddress, setInputAddress] = useState(location.address || "");
+  const [loadingPlaceDetails, setLoadingPlaceDetails] = useState(false);
+
+  const onInputChange = async (event) => {
```

Comment:
probably best to debounce this so it doesn't trigger for every character that's being typed 
https://github.com/react-hook-form/react-hook-form/issues/40

Reply:
Good call I mentioned it at some meeting but forgot to note it here. Should I use lodash or a specific react library https://www.npmjs.com/package/react-debounce-input?

Reply:
I've used lodash a lot before, it used to be very common as utility functions but nowadays a lot of its functionality is included in ECMASCript by default, it's still pretty good for some other things like `debounce` and it can import those utilities independently so for me that's ok.
If you can get that other component to work then it's also ok, but for form components we have currently a mix of libraries and our own custom components so that might be more difficult to add another one.

Reply:
I'm good with lodash. I'll do that now. The dropdown was a super easy fix I didn't see before. They have a visible prop that for some reason I didn't notice.

Reply:
I added `lodash/debounce`. The link you provided - the first few solutions didn't quite work and I didn't want to add that library - `awesome-debounce-promise` -  no commits for a year so dunno if it is being maintained. But adapting this worked for me: https://codesandbox.io/s/lodash-debounce-vlq1x?from-embed=&file=/src/index.js:1044-1054